### PR TITLE
Enable to set config file path when starting server

### LINKED ISSUE #141: Enable to set config file path when starting server
> Since this is a repo for general use, the system specific information should not be included. Also it would be better for security.
Let's make the config template just like iquip.

_Originally posted by @kangz12345 in https://github.com/snu-quiqcl/artiq-proxy/issues/140#issuecomment-2046373979_

### Feature you want to implement
Like IQUIP, I will enable to set the config file path in the command to start the server.

### How the feature is implemented
In FastAPI, [Setting](https://fastapi.tiangolo.com/advanced/settings/?h=#pydantic-settings) is supported to enable to set a config outside of the Python code.
Using this, I will add an enviroment variable, `config_path`, the target config file path.

### PR DESCRIPTION
**Pull request by BECATRUE**
_(updated)_

This closes #141.

Now, one can set the target config file path as follows:
```shell
$ CONFIG_PATH="config.json" ./server.sh           # target config file = "config.json"

$ CONFIG_PATH="config_template.json" ./server.sh  # target config file = "config_template.json"

$ ./server.sh                                     # target config file = default ("config.json")
```

For convenience, the command line arguments are supported in `server.sh`:
```shell
$ ./server.sh -c config.json           # target config file = "config.json"

$ ./server.sh -c config_template.json  # target config file = "config_template.json"

$ ./server.sh                          # target config file = default ("config.json")
```

The followings are example results:

### CODE REVIEW

**Comment by kangz12345** [comment]
Great! `BaseSetting` would enable more flexible parameterization.
As you said, however, setting the environment variables is not familiar to our usual cases.
How about exploiting the shell script to make it as a command line argument?
In other words, for example,
```shell
./server.sh -c config.json
```
and for its implementation, [a Stack Overflow answer](https://unix.stackexchange.com/a/505342) may help.

**Comment by BECATRUE** [comment]
I see! It looks much helpful. I will implement such the shell script.

**Comment by BECATRUE** [comment]
Please check the updated `server.sh`! @kangz12345 
(The re-require button has been disabled😅)

File: server.sh [hunk 1]
```diff
@@ -1 +1,23 @@
...
+while getopts "c:" opt
+do
+    case "$opt" in
+        c) config_path=$OPTARG ;;
+        *) echo "Invalid option" ; helpFunction ;;
+    esac
+done
+
+if [ -z "$config_path" ]; then
+    python -m uvicorn main:app --host 0.0.0.0 --port 8000
```
Comment (kangz12345) [inline server.sh]: How about defining variables for host address and port number in the shell script? For convenient modification of the script.

**Review by kangz12345 — CHANGES_REQUESTED** [review]

**Review by kangz12345 — APPROVED** [review]
LGTM!

**Pull request merged**